### PR TITLE
PF-82: Login for employees

### PR DESCRIPTION
**Pull request by tpambor**
## Summary by CodeRabbit

- **New Features**
	- Introduced `DashboardComponent`, `EmployeeListComponent`, and `IncidentListComponent` for enhanced analytics and employee management.
	- Implemented `LoginComponent` for user authentication with a reactive form.
	- Added role-based routing for improved access control.
	- Enhanced internationalization support with new translation units for error messages and UI elements.

- **Bug Fixes**
	- Updated internationalization handling for various components.

- **Documentation**
	- Improved translation units for Spanish and Portuguese, enhancing user experience.

- **Tests**
	- Added comprehensive test suites for new components, guards, and services, ensuring functionality and reliability.

- **Chores**
	- Updated dependencies and configuration files for better project management and performance.

### CODE REVIEW

File: src/app/analytics/dashboard/dashboard.component.ts [hunk 1]
```diff
@@ -0,0 +1,10 @@
+import { Component } from '@angular/core';
+
+@Component({
+  selector: 'app-dashboard',
+  standalone: true,
+  imports: [],
+  templateUrl: './dashboard.component.html',
+  styleUrl: './dashboard.component.scss',
+})
```
[the remark this answers is not in the document]
Comment (tpambor) [reply src/app/analytics/dashboard/dashboard.component.ts:9]: styleUrl is fine as well, if there is only one stylesheet

File: src/app/auth/login/login.component.ts [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+    });
+  }
+
+  get username() {
+    return this.loginForm.get('username')!;
+  }
+
+  get password() {
+    return this.loginForm.get('password')!;
+  }
```
[the remark this answers is not in the document]
Comment (tpambor) [reply src/app/auth/login/login.component.ts:56]: Form should be already initialized when these are called.

File: src/app/auth/auth.service.ts [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+      return null;
+    }
+
+    const decodedToken = jwtDecode(token);
+    const role = Object.values(Role).includes(decodedToken.aud as Role)
+      ? (decodedToken.aud as Role)
+      : null;
+
+    return role;
+  }
```
[the remark this answers is not in the document]
Comment (tpambor) [reply src/app/auth/auth.service.ts:59]: Verification happens server-side

File: src/app/auth/auth.guard.ts [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+export const authGuard: CanActivateFn = (route, _state) => {
+  const authService = inject(AuthService);
+  const router = inject(Router);
+
+  const role = authService.getRole();
+
+  if (!authService.isAuthenticated() || role === null) {
+    router.navigate(['/']);
+    return false;
+  }
```
[the remark this answers is not in the document]
Comment (tpambor) [reply src/app/auth/auth.guard.ts:15]: Valid token but unknown role is considered here invalid.

File: src/app/auth/auth.guard.ts [hunk 5]
```diff
@@ -0,0 +1,25 @@
...
+
+  const roles = route.data['roles'];
+
+  if (!roles.includes(role)) {
+    router.navigate([defaultRoutes[role]]);
+    return false;
+  }
```
[the remark this answers is not in the document]
Comment (tpambor) [reply src/app/auth/auth.guard.ts:22]: If the auth guard is used, role always have to be defined

File: src/app/auth/login/login.component.ts [hunk 6]
```diff
@@ -0,0 +1,75 @@
...
+    this.authService.login(this.username.value, this.password.value).subscribe(success => {
+      if (!success) {
+        return;
+      }
+
+      const role = this.authService.getRole();
+      if (role !== null) {
+        this.router.navigate([defaultRoutes[role]]);
+      }
+    });
```
[the remark this answers is not in the document]
Comment (tpambor) [reply src/app/auth/login/login.component.ts:73]: This is handled using a interceptor

File: src/locale/messages.pt-BR.xlf [hunk 7]
```diff
@@ -2,14 +2,74 @@
 <xliff version="1.2" xmlns="urn:oasis:names:tc:xliff:document:1.2">
   <file source-language="es-CO" datatype="plaintext" original="ng2.template">
```
[the remark this answers is not in the document]
Comment (tpambor) [reply src/locale/messages.pt-BR.xlf:3]: This file is generated automatically.

File: src/app/auth/login/login.component.ts [hunk 8]
```diff
@@ -0,0 +1,75 @@
...
+    private readonly router: Router,
+    private readonly formBuilder: FormBuilder,
+  ) {}
+
+  ngOnInit(): void {
+    if (this.authService.isAuthenticated()) {
+      const role = this.authService.getRole();
+      if (role !== null) {
+        this.router.navigate([defaultRoutes[role]]);
+      }
```
[the remark this answers is not in the document]
Comment (tpambor) [reply src/app/auth/login/login.component.ts:41]: If role is null, desired behavior is to stay on the login page

File: package.json [hunk 9]
```diff
@@ -24,6 +24,8 @@
     "@angular/platform-browser": "^18.2.8",
     "@angular/platform-browser-dynamic": "^18.2.8",
     "@angular/router": "^18.2.8",
+    "@faker-js/faker": "^9.0.3",
```
[the remark this answers is not in the document]
Comment (tpambor) [reply package.json:27]: It is used in the unit tests

**Review by VivianaOj — APPROVED** [review]

**Pull request merged**